src/wasm/wasm-binary.cpp

```diff
@@ -1253,6 +1348,12 @@ Type WasmBinaryBuilder::getType() {
       return Type::anyref;
     case BinaryConsts::EncodedType::eqref:
       return Type::eqref;
+    case BinaryConsts::EncodedType::nullable:
+      return Type(getHeapType(), /* nullable = */ true);
+    case BinaryConsts::EncodedType::nonnullable:
+      // FIXME: support non-nullable types. search for all "nullable = "
+      //        comments.
+      return Type(getHeapType(), /* nullable = */ true);
```

Comment:
I think it would be better to error out here rather than silently change the users' program.

Reply:
I thought the plan we agreed on was to just do everything with nullable types for now, which is definitely broken, but it's a stepping stone? Erroring out makes sense in production-ready code, I agree, but this is all very much wip.

Reply:
I agree that we should only support nullable types for now, but IMO that should include parsing as well. In general, this makes our level of support explicit to our users and will help us find the places we need to change later when we do want to support nullable types.

Reply:
Do you mean parsing of custom user types (not parsing of `ref.func` etc.)? I guess there's no harm to allow non-nullable types there for now, as long as we don't test them, sgtm.

Reply:
> I thought the plan we agreed on was to just do everything with nullable types for now.

I think we interpreted that differently. My interpretation was that we should only _support_ nullable types for now, which to me means erroring out when encountering a non-nullable type in the user input. It looks like your interpretation allows for "supporting" non-nullable types from the user's point of view by treating them as nullable types. I feel less strongly about this particular instance now because some of the implementation strategies we were brainstorming earlier did involve treating non-nullable types as nullable in Binaryen IR, but on a related note, do you plan to add validation that local, block, and function types do not contain non-nullable references in this PR?